gossip/gossip.go

```diff
@@ -352,6 +352,7 @@ func (g *Gossip) bootstrap() {
 		g.mu.Lock()
 		g.parseBootstrapAddresses()
 		if g.closed {
+			g.mu.Unlock()
```

Comment:
No, that code is correct. Waiting on a condition variable gives up the mutex. But when the wait ends, the mutex is again held.

http://golang.org/pkg/sync/#Cond.Wait